pkg/odo/cli/component/push.go

```diff
@@ -43,42 +45,73 @@ const PushRecommendedCommandName = "push"
 
 // PushOptions encapsulates options that push command uses
 type PushOptions struct {
-	ignores    []string
-	local      string
-	sourceType string
-	sourcePath string
-	*ComponentOptions
+	ignores          []string
+	local            string
+	sourceType       string
+	sourcePath       string
+	localConfig      *config.LocalConfigInfo
+	componentContext string
+	*genericclioptions.Context
+	client *occlient.Client
 }
 
 // NewPushOptions returns new instance of PushOptions
 func NewPushOptions() *PushOptions {
-	return &PushOptions{[]string{}, "", "", "", &ComponentOptions{}}
+	return &PushOptions{
+		ignores:     []string{},
+		localConfig: &config.LocalConfigInfo{},
+	}
 }
 
 // Complete completes push args
 func (po *PushOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
-	err = po.ComponentOptions.Complete(name, cmd, args)
-	if err != nil {
-		return err
+	po.client = genericclioptions.Client(cmd)
+
+	if len(po.componentContext) > 0 {
+		err = po.localConfig.GetLocalConfigFileFromPath(po.componentContext)
+		if err != nil {
+			return errors.Wrapf(err, "failed to access component config from %+v", po.componentContext)
+		}
+
+		err = po.localConfig.Loadconfig()
+		if err != nil {
+			return errors.Wrapf(err, "failed to load compoennt settings from %s", po.componentContext)
+		}
+	} else {
+		conf, err := config.NewLocalConfigInfo()
+		if err != nil {
+			return errors.Wrap(err, "failed to fetch component config")
+		}
+		po.localConfig = conf
```

Comment:
This above section, I'm just implementing https://github.com/redhat-developer/odo/issues/1419

We should separate the above code into a separate function that retrieves the configuration and returns it. 

We are doing two things:
1. We load from passing `--context`
2. We load from the current directory.

This functionality is going to happen in A LOT of other functions..

Reply:
Done NewLocalConfigInfo now takes a filepointer :)

Reply:
and context is anyway a command level flag so can't be generalized further in my opinion